doc/designs/quic-design/rx-depacketizer.md

```diff
@@ -0,0 +1,257 @@
+RX depacketizer
+===============
+
+This component takes a QUIC packet and parses the frames contained therein,
+to be forwarded to appropriate other components for further processing.
+
+In the [overview], this is called the "RX Frame Handler".  The name "RX
+depacketizer" was chosen to reflect the kinship with the [TX packetizer].
+
+Structures
+----------
+
+### Connection
+
+Represented by an SSL object.
+
+### Stream
+
+Represented by an SSL object.
+
+### Packets
+
+Represented by the `OSSL_QUIC_PACKET` structure.
+
+*This structure is inspired from the structure with the same name presented
+with the [TX packetizer], but takes into account the needs of the RX
+depacketizer as well.*
+
+It's expected that the QUIC Read Record Layer decrypts the packet and
+parses its packet header, and creates the `OSSL_QUIC_PACKET` structure with
+appropriately populated fields.
+
+The RX depacketizer depends on `packet_data`, `packet_data_length` and
+`packet_payload_offset` to find the packet payload and parse frames out of
+it.  Additionally, it uses `connection`, `packet_number` and `packet_type`.
+
+It's assume that `packet->connection` can be used to find the appropriate
+stream SSL objects attached to the connection.
+
+```c
+struct ossl_quic_packet_st {
+    SSL *connection;
```

Comment:
This should be a QUIC_CONNECTION if the refactoring PR is merged.

Reply:
N/A.  I'm now referring to #18870